streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -620,6 +650,11 @@
                     "",
                     Importance.LOW,
                     APPLICATION_SERVER_DOC)
+            .define(BALANCE_FACTOR_CONFIG,
+                    Type.INT,
+                    1,
```

Comment:
req: Please add `atLeast(1)` validator here and the corresponding unit test. Similar applies also to the other configs.